CONTRIBUTING.md

```diff
@@ -61,7 +62,76 @@ about [how to report a new issue](#reporting-an-issue)!
     This is important to avoid the overlapping of work!
 
 1. If the issue is already in progress, feel free to ask questions or engage in conversation about
-    the implementation. Otherwise, continue on to [working on issues](#working-on-issues).
+    the implementation. Otherwise, review the repository license requirements
+    before starting work and continue on to [working on issues](#working-on-issues).
+
+### Reviewing license requirements
+
+To contribute to any CyberArk open source project, you must make sure you've met
+our contributor license requirements.
+
+To find out what the project you're interested in requires, refer to the
+`CONTRIBUTING.md` file in the repository.
+
+#### When the Repo Includes the CLA
+
+If the `CONTRIBUTING.md` document says you need to sign the CLA, for example:
+
+> "Thanks for your interest in Conjur. Before contributing, please take a moment to read and sign
+> our Contributor Agreement. This provides patent protection for all Conjur users and allows
+> CyberArk to enforce its license terms. Please email a signed copy to oss@cyberark.com."
+
+Then you **must** submit a signed copy of the CLA to oss@cyberark.com in order
+for your contribution to be merged.
+
+A copy of the CLA can be found [here](/documents/CyberArk_Open_Source_Contributor_Agreement.pdf).
+
+#### When the Repo Does Not Include the CLA
+
+If the repo does not ask you to sign the CLA, then you may do one of the following:
+
+-  Follow the above instructions for [signing the CLA](#when-the-repo-includes-the-cla).
+
+**OR**
+
+-  Sign your commits using the `--signoff` flag in the `git` CLI (more info
+   [here](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s)) to certify that you
+   have the rights to submit the work under the same license as the project and you agree to a
+   Developer Certificate of Origin (see http://developercertificate.org/ for more information).
+
+##### Signing Off a Commit
+
+When signing, every commit must be signed, and your PR will not be approved until this is the case.
+
+To sign-off, use:
+
+    git commit --signoff
```

Comment:
We should use the escape ticks instead of indents for code snippets. Same comment for code right below.